### PR TITLE
Replace key registry with node registry

### PR DESCRIPTION
**Pull request by ltseeley**
Removes the key registry from splinter and merges its functionality with the node registry

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,12 +32,12 @@
 
 /examples/private_xo/target
 /examples/private_xo/Cargo.lock
-/examples/private_xo/demo_state/keys.yaml
```
Comment (agunde406) [inline .gitignore:35]: typo in commit message: splitnerd -> splinterd

File: libsplinter/src/admin/service/shared.rs [hunk 2]
```diff
@@ -1034,31 +1035,36 @@ impl AdminServiceShared {
...
+                    "Requester node {} not found in node registry",
+                    requester_node_id,
                 ))
             })?;
 
-        if key_info.associated_node_id() != requester_node_id {
+        if !requester_node.has_key(&to_hex(signer_public_key)) {
             return Err(AdminSharedError::ValidationFailed(format!(
-                "{} is not registered for the node in header",
+                "{} is not registered for the requester node",
```
[the remark this answers is not in the document]
Comment (ltseeley) [reply libsplinter/src/admin/service/shared.rs]: Added

File: libsplinter/src/admin/service/shared.rs [hunk 3]
```diff
@@ -1277,28 +1283,30 @@ impl AdminServiceShared {
...
 
-        let signer_node = key_info.associated_node_id().to_string();
-
-        if signer_node != node_id {
+        if !node.has_key(&to_hex(signer_public_key)) {
             return Err(AdminSharedError::ValidationFailed(format!(
-                "Payload requester {} is not registered to node {}",
-                to_hex(circuit_proposal.get_requester()),
-                node_id
+                "{} is not registered for the voting node",
```
[the remark this answers is not in the document]
Comment (ltseeley) [reply libsplinter/src/admin/service/shared.rs]: Fixed

File: .gitignore (hunk 1 above)
Comment (ltseeley) [reply .gitignore:35]: Fixed

File: examples/gameroom/docker-compose.yaml [hunk 4]
```diff
@@ -133,24 +163,17 @@ services:
...
-          done && \
-          # Copy the generated key registry to its expected location
-          cp -a /key_registry_shared/keys.yaml /var/lib/splinter && \
           splinter cert generate --skip && \
           splinterd -c ./configs/splinterd-node-acme.toml -vv \
               --service-endpoint tcp://0.0.0.0:8043 \
               --network-endpoints tcps://0.0.0.0:8044 \
+              --advertised-endpoints tcps://splinterd-node-acme:8044 \
               --bind 0.0.0.0:8085 \
+              --registry http://registry-server:80/nodes.yaml \
```
Comment (agunde406) [inline examples/gameroom/docker-compose.yaml:176]: --registry will work but it has been updated to --registries

Comment (ltseeley) [reply examples/gameroom/docker-compose.yaml]: Fixed

**Pull request merged**